src/main/java/org/ethereum/beacon/discovery/util/Utils.java

```diff
@@ -97,4 +97,15 @@ public static UInt64 toUInt64(Bytes bytes) throws IllegalArgumentException {
     checkArgument(bytes.size() <= 8);
     return UInt64.fromBytes(Utils.leftPad(bytes, 8));
   }
+
+  public static int compareBytes(Bytes b1, Bytes b2) {
+    checkArgument(b1.size() == b2.size(), "Comparing bytes of different sizes not supported");
```

Comment:
Should we just return false if the bytes are of different sizes?

Reply:
What do you mean under `false` ? For `equals()` we would return `false`, but for `compare()` we should normally return different results when swapping arguments.
Since this method is used just for `Bytes32` I just changed param types and removed the check

Reply:
ah - nm, I was thinking this was an equality check